packages/task/src/browser/task-service.ts

```diff
@@ -947,7 +967,9 @@ export class TaskService implements TaskConfigurationClient {
             }
         );
         this.shell.addWidget(widget, { area: 'bottom' });
-        this.shell.activateWidget(widget.id);
+        if (taskInfo && taskInfo.config.presentation && taskInfo.config.presentation.reveal === RevealKind.Always) {
```

Comment:
Should not we drop `&& taskInfo.config.presentation.reveal === RevealKind.Always`?

Reply:
i updated the logic in `showTaskTerminal()` in the latest patch. With the change of logic `&& taskInfo.config.presentation.reveal === RevealKind.Always` should stay now.

Reply:
i don't think it is good, now it is not correct at https://github.com/eclipse-theia/theia/pull/6814/files#diff-794cdf6ef93aef7cc827f31f92ebe226R678

I think before it was better that silent should be checked explicitly, now if someone call `showTaskTerminal` with silent without checking it will be a bug

Reply:
maybe I should change the `showTaskTerminal()` into the following
```
showTaskTerminal(terminal, focus) {
  if (focus) { this.shell.activateWidget(terminal); }
  else { this.shell.revealWidget(terminal); }
}
```
This way I can explicitly leave the `reveal = always/silent/never` to the caller of `showTaskTerminal()`. What do you think? Thank you !

Reply:
focus does not seem to be an issue, it is always based on presentation.focus, but in else case whether something should be revealed is different from case to case, maybe pass reveal variable instead?

Reply:
`focus` looks important to me. When a user starts a task that is already active, it is `presentation.focus` that decides if `shell.activateWidget` or `shell.revealWidget` should be called, is it?